File: php/testcode/codeinj_032.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_codeinj_wakeup_gadget
class EvalGadget032 {
    public string $code = '';
    public function __wakeup(): void {
        eval($this->code);
    }
}

function codeinj032(BenchmarkRequest $req): BenchmarkResponse {
    $data = $req->post('data');
    $obj = unserialize($data); // vuln-code-snippet vuln-line php_codeinj_wakeup_gadget
    return BenchmarkResponse::ok('deserialized');
}
// vuln-code-snippet end php_codeinj_wakeup_gadget

File: ruby/testcode/codeinj_032.rb
require_relative 'shared'

# vuln-code-snippet start ruby_codeinj_frozen_bypass
def execute_snippet(req)
  frozen_input = req.param('code').freeze
  unfrozen = frozen_input.dup
  result = eval(unfrozen) # vuln-code-snippet vuln-line ruby_codeinj_frozen_bypass
  BenchmarkResponse.json({ result: result.to_s })
end
# vuln-code-snippet end ruby_codeinj_frozen_bypass
